Authentication Flow

vanyalaptii opened the pull request:
Created logic for:
1. SignIn.
2. SignUp.
3. LogOut.
4. Account delete(just authentication).
5. Update and reset password.
6. Simple basic validation for email, password and confirmPassword text fields.
7. ProfileView and ProfileViewModel added

vanyalaptii committed:
Authentication Flow

Created logic for:
1. SignIn.
2. SignUp.
3. LogOut.
4. Account delete(just authentication).
5. Update and reset password.
6. Simple basic validation for email, password and confirmPassword text fields.
7. ProfileView and ProfileViewModel added

OksanaFedorchuk commented on ReFix/Views/MainView.swift:
```diff
@@ -8,31 +8,47 @@
+            }
+        }
+        .onAppear {
+            let authUser = try?  AuthenticationManager.shared.getAuthenticatedUser()
```
це перенести в init вю моделі

OksanaFedorchuk commented on ReFix/Modules/AuthView/AuthView.swift:
```diff
@@ -8,18 +8,18 @@
 struct AuthView: View {
-    
-    @StateObject private var viewModel = AuthViewModel()
+    @Binding var showAuthView: Bool
```
прибрати це проперті, презентація в"ю менеджиться тим хто його презентує, в"ю не пеовинен про це знати

OksanaFedorchuk commented on ReFix/Modules/AuthView/AuthenticationManager.swift:
```diff
@@ -0,0 +1,77 @@
+import Foundation
+import FirebaseAuth
+
+struct AuthDataResultModel {
```
чому модель лежить в файлі менеджера?

OksanaFedorchuk commented on ReFix/Views/MainView.swift:
```diff
@@ -8,31 +8,47 @@
+            authViewModel.showAuthView = authUser == nil ? true : false
+        }
+        .fullScreenCover(isPresented: $authViewModel.showAuthView) {
+            NavigationStack {
```
навіщо тут стек?

OksanaFedorchuk commented on ReFix/ReFixApp.swift:
```diff
@@ -11,12 +11,19 @@ import Firebase
+class AppDelegate: NSObject, UIApplicationDelegate {
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? =
+                     nil) -> Bool {
+        FirebaseApp.configure ()
```
FirebaseApp.configure()

vanyalaptii committed:
Remove .onAppear modifier

1. Remove .onAppear modifier from MainView.
2. Move AuthDataResultModel to separate file.
3. Remove NavigationStack from .fullScreenCover modifier in MainView.

vanyalaptii committed:
Fixed Authentication logic.

Fixed Authentication logic.

vanyalaptii committed:
Tab focus added.

Tab focus added.

OksanaFedorchuk commented on ReFix/Modules/AuthView/AuthenticationManager.swift:
```diff
@@ -0,0 +1,72 @@
+    static let shared = AuthenticationManager()
+    private init(){}
+    
+    func isHaveLoggedUser() -> Bool {
```
зараз в тебе цей менеджер, який займається роботою з базою даних в фаєрбейсі, тримає значення чи залогінений юзер чи ні. Це треба перенести в метод AuthModel, під час якого призначати Published property 
@Published var isUserLoggedIn: Bool = false (замість @Published var showAuthView: Bool)
Сам метод запускати в init() AuthModel

OksanaFedorchuk commented on ReFix/Models/AuthDataResultModel.swift:
```diff
@@ -0,0 +1,19 @@
+import Foundation
+import Firebase
+
+struct AuthDataResultModel {
```
UserModel?

OksanaFedorchuk commented on ReFix/Modules/AuthView/SignInView.swift:
```diff
@@ -9,6 +9,7 @@ import SwiftUI
 
 struct SignInView: View {
     
+//    @Binding var showAuthView: Bool
```
це ще потрібно?

vanyalaptii committed:
isUserLoggedIn, FocusState, Alerts

1. The isUserLoggedIn property has been added instead of showAuthScreen.
2. isFocused property type is changed  to Field? instead of Bool.
3. added alerts for a autentication validations.

vanyalaptii committed:
findLoggedUser(), switch

1. findLoggedUser() used in AuthViewModel init()
2. Using "switch" instead to "if else" for toggle main and authentification screens.
